core/src/main/scala/org/apache/spark/util/ShutdownHookManager.scala

```diff
@@ -177,8 +181,19 @@ private [util] class SparkShutdownHookManager {
     val hookTask = new Runnable() {
       override def run(): Unit = runAll()
     }
-    org.apache.hadoop.util.ShutdownHookManager.get().addShutdownHook(
-      hookTask, FileSystem.SHUTDOWN_HOOK_PRIORITY + 30)
+    val priority = FileSystem.SHUTDOWN_HOOK_PRIORITY + 30
+    // The timeout property must be passed as a Java system property because this
+    // is initialized before Spark configurations are registered as system
+    // properties later in initialization.
+    val timeout = new SparkConf().get(SPARK_SHUTDOWN_TIMEOUT_MS)
+
+    timeout.fold {
+      org.apache.hadoop.util.ShutdownHookManager.get().addShutdownHook(
+        hookTask, priority)
+    } { t =>
+      org.apache.hadoop.util.ShutdownHookManager.get().addShutdownHook(
```

Comment:
Just a question. It seems that we invoke `org.apache.hadoop.util.ShutdownHookManager.get()` three times repeatedly. Is this inevitable?

Reply:
I only see two places `org.apache.hadoop.util.ShutdownHookManager.get()` exists. Can you clarify where the third location is? Only one case will be invoked. The case on line 191 is if the option is empty and line 194 is if the option is not emoty.

Reply:
Ah, you're right. My bad. I mistakenly counted `line 180` which is a deleted line.